shared/middleware/net_sim_api.rb

```diff
@@ -178,32 +178,94 @@ def owns_shard?(shard_id)
   #
   # POST /v3/netsim/<shard-id>/<table-name>
   #
-  # Insert a new row.
+  # Insert a new row or rows.
+  # Accepts a JSON request body of either an Object if inserting a
+  # single row or an Array of Objects if inserting multiple rows
+  #
+  # Examples:
+  #
+  # POST "{ 'name': 'alice', 'male': false }"
+  # -> 201 "{ 'name': 'alice', 'male': false, 'id': 1 }"
+  #
+  # POST "[{ 'name': 'nancy', 'male': false }, { 'name': 'drew', 'male': true }]"
+  # -> 201 "[{ 'name': 'nancy', 'male': false, 'id': 2 }, { 'name': 'drew', 'male': true, 'id': 3 }]"
   #
   post %r{/v3/netsim/([^/]+)/(\w+)$} do |shard_id, table_name|
     dont_cache
     unsupported_media_type unless has_json_utf8_headers(request)
 
+    # Parse JSON
     begin
-      value = get_table(shard_id, table_name).
-          insert(JSON.parse(request.body.read), request.ip)
-      if table_name == TABLE_NAMES[:message]
-        node_exists = get_table(shard_id, TABLE_NAMES[:node]).to_a.any? do |node|
-          node['id'] == value['simulatedBy']
-        end
-        unless node_exists
-          get_table(shard_id, table_name).delete(value['id'])
-          json_bad_request
-        end
-      end
+      body = JSON.parse(request.body.read)
     rescue JSON::ParserError
       json_bad_request
     end
 
+    # Determine whether or not we are performing a multi-insert and
+    # normalize our request body into an array of values
+    if body.is_a?(Array)
+      multi_insert = true
```

Comment:
I wonder if we'd be better off just requiring that all inserts be formed as multi-inserts, and sending an array of one any time we only have one row to insert - less branching, less 'surface area' so to speak.  Probably not much added cost.

I say this hypocritically because I haven't finished this clean-up with deletes, myself.  I think the client is only using multi-delete at this point but I haven't torn out the single-delete endpoint from the API.

@philbogle thoughts?